web/api/v1/api.go

```diff
@@ -882,6 +882,9 @@ func (api *API) series(r *http.Request) (result apiFuncResult) {
 	warnings := set.Warnings()
 
 	for set.Next() {
+		if err := ctx.Err(); err != nil {
+			return apiFuncResult{nil, contextErr(err), nil, nil}
```

Comment:
Better to still return warnings and the closer?

```
return apiFuncResult{nil, contextErr(err), warnings, closer}
```

Reply:
thanks, added the warning and closer.